internal/cluster/connection_manager.go

```diff
@@ -613,32 +634,42 @@ func (m *connectionMap) FindAddedAddrs(members []pubcluster.MemberInfo, cs *Serv
 			addedAddrs = append(addedAddrs, addr)
 		}
 	}
-	m.connectionsMu.RUnlock()
+	m.mu.RUnlock()
 	return addedAddrs
 }
 
 func (m *connectionMap) FindRemovedConns(members []pubcluster.MemberInfo) []*Connection {
-	m.connectionsMu.RLock()
+	m.mu.RLock()
 	removedConns := []*Connection{}
 	for _, member := range members {
 		addr := member.Address
 		if conn, exists := m.addrToConn[addr]; exists {
 			removedConns = append(removedConns, conn)
 		}
 	}
-	m.connectionsMu.RUnlock()
+	m.mu.RUnlock()
 	return removedConns
 }
 
 func (m *connectionMap) Info(infoFun func(connections map[pubcluster.Address]*Connection, connToAddr map[int64]pubcluster.Address)) {
-	m.connectionsMu.RLock()
+	m.mu.RLock()
 	infoFun(m.addrToConn, m.connToAddr)
-	m.connectionsMu.RUnlock()
+	m.mu.RUnlock()
 }
 
 func (m *connectionMap) Len() int {
-	m.connectionsMu.RLock()
+	m.mu.RLock()
 	l := len(m.connToAddr)
-	m.connectionsMu.RUnlock()
+	m.mu.RUnlock()
 	return l
 }
+
+func (m *connectionMap) removeAddr(addr pubcluster.Address) {
+	for i, a := range m.addrs {
+		if a.Equal(addr) {
+			// note that this changes the order of addresses
+			m.addrs[len(m.addrs)-1] = m.addrs[i]
+			m.addrs = m.addrs[:len(m.addrs)-1]
```

Comment:
does not this always remove the last connection? You set the last connection to be m.addrs[i] then you discard the last element.

Reply:
Good catch, fixed at: https://github.com/hazelcast/hazelcast-go-client/pull/580/commits/8ff2f4a9dc5ac6228040a9b24a7045c40fa06bcd